### PR TITLE
Beautify the `poll()` system call

### PR DESCRIPTION
**Pull request by lrh2000**
This is a simple cleanup PR that splits the large `do_poll` method into several smaller methods.

Only for code beauty and readability, no functionality or performance changes expected.

### CODE REVIEW

**Review by StevenJiang1110 — COMMENTED** [review]
Good job!

File: kernel/src/syscall/poll.rs [hunk 1]
```diff
@@ -97,7 +81,77 @@ pub fn do_poll(poll_fds: &[PollFd], timeout: Option<Duration>, ctx: &Context) ->
...
+
+    let mut files = Vec::with_capacity(poll_fds.len());
+
+    for poll_fd in poll_fds.iter() {
+        let Some(fd) = poll_fd.fd() else {
+            files.push(None);
+            continue;
+        };
+
+        files.push(Some(file_table.get_file(fd)?.clone()));
```
[the remark this answers is not in the document]
Comment (lrh2000) [reply kernel/src/syscall/poll.rs:104]: Ah, interesting. I think I checked the behavior for epoll some time ago, and I _didn't_ get `POLLINVAL` for epoll (if I did the right experiments and remembered the results correctly). So I _assume_ that `poll()` would do the same thing.

I'll take a look at the scenario. Thanks for the information!

[the remark this answers is not in the document]
Comment (lrh2000) [reply kernel/src/syscall/poll.rs]: > If so, the `hold_files` method wouldn’t need to return a `Result`.

Actually it needs to return something to indicate whether or not it found an invalid file descriptor. See the new commit for details.

**Review by StevenJiang1110 — APPROVED** [review]
LGTM!

**Pull request merged**